components/common/Input.tsx

```diff
@@ -0,0 +1,80 @@
+import styled from 'styled-components';
+import eyeOffIcon from '@/public/assets/images/eye-off.svg';
+import eyeOnIcon from '@/public/assets/images/eye-on.svg';
+import Image from 'next/image';
+import { useState } from 'react';
+
+interface Props {
+  passwordMode: boolean;
+  errorMessage: string;
+  placeholder: string;
+  focusOutHandler: () => void;
+}
+
+export default function Input({ passwordMode, errorMessage = '', placeholder, focusOutHandler }: Props) {
+  const [password, setPassword] = useState(true);
+
+  function handlePwToggle() {
+    setPassword((prev) => !prev);
+  }
+
+  return (
+    <>
+      <Container>
+        <InputWrapper onBlur={focusOutHandler} placeholder={placeholder} type={password ? 'password' : ''} className={errorMessage ? 'error' : ''} />
```

Comment:
```suggestion
        <InputWrapper onBlur={focusOutHandler} placeholder={placeholder} {...(password && { type: 'password' })}  {...(errorMessage && { className: 'error' })}/>
```

음 이게 더 깔끔한가..? ㅎㅎ 이렇게 넣어주면 실제로 password, errorMessage가 있을때에만 props 를 추가해줄수있어요.
뭐가 더 깔끔해보이는지는 애매하긴한데, 그래도 빈문자열로 넣는것보단 이게 나을것같기도하네요? ㅎㅎ